src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java

```diff
@@ -424,4 +439,31 @@ protected boolean useDefaultValueForUnknownEnum(DeserializationContext ctxt) {
           && (Boolean.TRUE.equals(_useDefaultValueForUnknownEnum)
           || ctxt.isEnabled(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE));
     }
+
+    // @since 2.15
+    protected Boolean _getUseEnumNaming() {
+        if (_useEnumNaming == null) {
+            _useEnumNaming = _valueClass.getAnnotation(EnumNaming.class) != null;
+        }
+        return _useEnumNaming;
+    }
+
+    // @since 2.15
+    protected EnumNamingStrategy _getNamingStrategy() {
+        EnumNamingStrategy namingStrategy = _namingStrategy;
+        if (namingStrategy == null) {
+            synchronized (this) {
+                namingStrategy = _namingStrategy;
+                if (namingStrategy == null) {
+                    EnumNaming enumNamingAnnotation = _valueClass.getAnnotation(EnumNaming.class);
```

Comment:
No; all access to annotations should go via `AnnotationIntrospector`, not check directly.